bug fix

halfmanhalffish opened the pull request:
when quitting the bufexplorer window using the 'q' mapping, switch to the previously active buffer rather than the first buffer in the list also, preserve the correct alt buffer so that <C-^> works.

The bug was as follows:

I have 3 files open.  The first buffer (ie buffer with the lowest buffer number) is the alternate buffer, ie hidden and the one that I was editing before switching to the current buffer.  The second buffer is the current buffer.

I go into bufexplorer.  I decide that I didn't want to switch buffer after all, so I leave the cursor on the line where it was to start with (ie the current buffer) and I press return.  I expect to return to the current buffer but to my surprise I land in the alternate buffer.

If I am editing the first buffer, go into bufexplorer, change mind and hit return then I land in the first buffer as expected.

I uninstalled almost all my plugins and retested to see if that made a difference, and it didn't.

halfmanhalffish commented:
Jeff,

I submitted this pull request at a time before I realised what the
underlying problem was (ie spurious SessionLoadPost events in vim -
vim's fault, not BufExplorer's).  On my own machine I reverted my
changes (ie this merge, #15) and instead commented out the line

autocmd SessionLoadPost * call s:Reset()

in the function s:Setup() in BufExplorer.  That's a fine solution for
me since I don't use Vim sessions.

Which is the best solution for BufExplorer though?  I don't really
know enough about the subject.  My inclination would be to revert this
merge (#15) and instead to have a warning about SessionLoadPost in the
documentation.  Perhaps disable the SessionLoadPost autocmd by default
and leave it as an option.

What do you think ?

Philip (aka halfmanhalffish)

On 14/01/2015, jlanzarotta <notifications@github.com> wrote:
> Merged #15.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/jlanzarotta/bufexplorer/pull/15#event-218738870

jlanzarotta commented:
I am really struggling with git.  I am use to subversion; however, git
is somewhat different...

I believe I reverted your change.  I do not use Sessions either. I agree
that a warning in the documentation should be sufficient.

On 1/20/2015 8:06 AM, halfmanhalffish wrote:
> Jeff,
> 
> I submitted this pull request at a time before I realised what the
> underlying problem was (ie spurious SessionLoadPost events in vim -
> vim's fault, not BufExplorer's).  On my own machine I reverted my
> changes (ie this merge, #15) and instead commented out the line
> 
> autocmd SessionLoadPost * call s:Reset()
> 
> in the function s:Setup() in BufExplorer.  That's a fine solution for
> me since I don't use Vim sessions.
> 
> Which is the best solution for BufExplorer though?  I don't really
> know enough about the subject.  My inclination would be to revert this
> merge (#15) and instead to have a warning about SessionLoadPost in the
> documentation.  Perhaps disable the SessionLoadPost autocmd by default
> and leave it as an option.
> 
> What do you think ?
> 
> Philip (aka halfmanhalffish)
> 
> On 14/01/2015, jlanzarotta <notifications@github.com> wrote:
>> Merged #15.
>>
>> ---
>> Reply to this email directly or view it on GitHub:
>> https://github.com/jlanzarotta/bufexplorer/pull/15#event-218738870
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/jlanzarotta/bufexplorer/pull/15#issuecomment-70650260
>

halfmanhalffish commented:
Hi Jeff,

I recommend git.  It does take some getting used to, but in fact it's
just a consistent application of one simple principle (ie that it's
all about the merge).

Anyway, it looks like my change is still in there.  There's one more
merge that's been done since the troublesome #15, so what I'd suggest
is to remove that commit (pull request #22), remove pull request #15
and then reapply pull request #22.

I can give some pointers with git commands if you like - let me know.

Philip

On 20/01/2015, jlanzarotta <notifications@github.com> wrote:
> I am really struggling with git.  I am use to subversion; however, git
> is somewhat different...
>
> I believe I reverted your change.  I do not use Sessions either. I agree
> that a warning in the documentation should be sufficient.
>
> On 1/20/2015 8:06 AM, halfmanhalffish wrote:
>> Jeff,
>>
>> I submitted this pull request at a time before I realised what the
>> underlying problem was (ie spurious SessionLoadPost events in vim -
>> vim's fault, not BufExplorer's).  On my own machine I reverted my
>> changes (ie this merge, #15) and instead commented out the line
>>
>> autocmd SessionLoadPost * call s:Reset()
>>
>> in the function s:Setup() in BufExplorer.  That's a fine solution for
>> me since I don't use Vim sessions.
>>
>> Which is the best solution for BufExplorer though?  I don't really
>> know enough about the subject.  My inclination would be to revert this
>> merge (#15) and instead to have a warning about SessionLoadPost in the
>> documentation.  Perhaps disable the SessionLoadPost autocmd by default
>> and leave it as an option.
>>
>> What do you think ?
>>
>> Philip (aka halfmanhalffish)
>>
>> On 14/01/2015, jlanzarotta <notifications@github.com> wrote:
>>> Merged #15.
>>>
>>> ---
>>> Reply to this email directly or view it on GitHub:
>>> https://github.com/jlanzarotta/bufexplorer/pull/15#event-218738870
>>
>> ---
>> Reply to this email directly or view it on GitHub:
>> https://github.com/jlanzarotta/bufexplorer/pull/15#issuecomment-70650260
>>
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/jlanzarotta/bufexplorer/pull/15#issuecomment-70653918

jlanzarotta commented:
Hi Philip,

I think I committed the changes to git.  I am using the command line git
on windows and it is quite a pain...  We will see if this worked...

-Jeff

On 1/21/2015 7:09 AM, halfmanhalffish wrote:
> Hi Jeff,
> 
> I recommend git.  It does take some getting used to, but in fact it's
> just a consistent application of one simple principle (ie that it's
> all about the merge).
> 
> Anyway, it looks like my change is still in there.  There's one more
> merge that's been done since the troublesome #15, so what I'd suggest
> is to remove that commit (pull request #22), remove pull request #15
> and then reapply pull request #22.
> 
> I can give some pointers with git commands if you like - let me know.
> 
> Philip
> 
> On 20/01/2015, jlanzarotta <notifications@github.com> wrote:
>> I am really struggling with git.  I am use to subversion; however, git
>> is somewhat different...
>>
>> I believe I reverted your change.  I do not use Sessions either. I agree
>> that a warning in the documentation should be sufficient.
>>
>> On 1/20/2015 8:06 AM, halfmanhalffish wrote:
>>> Jeff,
>>>
>>> I submitted this pull request at a time before I realised what the
>>> underlying problem was (ie spurious SessionLoadPost events in vim -
>>> vim's fault, not BufExplorer's).  On my own machine I reverted my
>>> changes (ie this merge, #15) and instead commented out the line
>>>
>>> autocmd SessionLoadPost * call s:Reset()
>>>
>>> in the function s:Setup() in BufExplorer.  That's a fine solution for
>>> me since I don't use Vim sessions.
>>>
>>> Which is the best solution for BufExplorer though?  I don't really
>>> know enough about the subject.  My inclination would be to revert this
>>> merge (#15) and instead to have a warning about SessionLoadPost in the
>>> documentation.  Perhaps disable the SessionLoadPost autocmd by default
>>> and leave it as an option.
>>>
>>> What do you think ?
>>>
>>> Philip (aka halfmanhalffish)
>>>
>>> On 14/01/2015, jlanzarotta <notifications@github.com> wrote:
>>>> Merged #15.
>>>>
>>>> ---
>>>> Reply to this email directly or view it on GitHub:
>>>> https://github.com/jlanzarotta/bufexplorer/pull/15#event-218738870
>>>
>>> ---
>>> Reply to this email directly or view it on GitHub:
>>> https://github.com/jlanzarotta/bufexplorer/pull/15#issuecomment-70650260
>>>
>>
>> ---
>> Reply to this email directly or view it on GitHub:
>> https://github.com/jlanzarotta/bufexplorer/pull/15#issuecomment-70653918
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/jlanzarotta/bufexplorer/pull/15#issuecomment-70826320
>

PhilRunninger commented:
It looks like pull request #15 is still in master. I can't say for sure that this will work, but if you use the web interface, it may work better. There is a Revert button on [this page](https://github.com/jlanzarotta/bufexplorer/pull/15) whose tooltip says it will "create a new pull request to revert these changes". Give that a try, and see what happens. It should revert only those changes, and leave the ones from #22 alone.

halfmanhalffish commented:
Hi,

I clicked the Revert button that Phil mentioned and it has created a
new pull request (number #24) that looks like it will do the right
thing  (it showed me the patch and it looked right).

Philip

On 21/01/2015, Phil Runninger <notifications@github.com> wrote:
> It looks like pull request #15 is still in master. I can't say for sure that
> this will work, but if you use the web interface, it may work better. There
> is a Revert button on [this
> page](https://github.com/jlanzarotta/bufexplorer/pull/15) whose tooltip says
> it will "create a new pull request to revert these changes". Give that a
> try, and see what happens. It should revert only those changes, and leave
> the ones from #22 alone.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/jlanzarotta/bufexplorer/pull/15#issuecomment-70855124
